diff -Nru frisbee-snapshot-20040312/frisbee/client.c frisbee-snapshot-orbit/frisbee/client.c
--- frisbee-snapshot-20040312/frisbee/client.c	2005-08-24 12:00:05.000000000 -0400
+++ frisbee-snapshot-orbit/frisbee/client.c	2005-08-24 11:38:56.000000000 -0400
@@ -687,18 +687,20 @@
 			    (wasidle*idledelay) / 1000000,
 			    ((wasidle*idledelay) % 1000000) / 1000);
 		else {
+		  int progress = (int)(100 * (TotalChunkCount - chunkcount) / TotalChunkCount);
+		  if (progress % 10 == 0) {
+		    // we should fall in here whenever 10 percent of chunks have been 
+		    // processed.
 			struct timeval estamp;
 
 			gettimeofday(&estamp, 0);
 			estamp.tv_sec -= stamp.tv_sec;
 		
-			printf(".");
+
+			printf("Progress: %d%% %.6d %.6d\n",
+			        progress, (int)estamp.tv_sec, chunkcount);
 			fflush(stdout);
-			if (dotcol++ > 65) {
-				dotcol = 0;
-				printf("%4ld %6d\n",
-				       estamp.tv_sec, chunkcount);
-			}
+		  }
 		}
 
 		CLEVENT(1, EV_CLIWRSTART,
diff -Nru frisbee-snapshot-20040312/frisbee/log.c frisbee-snapshot-orbit/frisbee/log.c
--- frisbee-snapshot-20040312/frisbee/log.c	2005-08-24 12:00:05.000000000 -0400
+++ frisbee-snapshot-orbit/frisbee/log.c	2005-08-24 11:38:56.000000000 -0400
@@ -71,8 +71,10 @@
 	va_end(args);
 
 	if (!usesyslog) {
+		/*
 		fputs(buf, stderr);
 		fputc('\n', stderr);
+		*/
 	}
 	else
 		syslog(LOG_INFO, "%s", buf);
diff -Nru frisbee-snapshot-20040312/imagezip/version.c frisbee-snapshot-orbit/imagezip/version.c
--- frisbee-snapshot-20040312/imagezip/version.c	1969-12-31 19:00:00.000000000 -0500
+++ frisbee-snapshot-orbit/imagezip/version.c	2005-08-24 11:38:57.000000000 -0400
@@ -0,0 +1 @@
+char build_info[] = "Built 11-Feb-2005 by root@internal2:/home/max/pxe/build/frisbee-snapshot-20040312/imagezip";
